src/main/java/org/jitsi/jigasi/JvbConference.java

```diff
@@ -1332,13 +1333,26 @@ private void inviteFocus(final EntityBareJid roomIdentifier)
             this.errorLog = errorLog;
         }
 
+        /**
+         * Whether the timeout had been scheduled.
+         * @return <tt>true</tt> if timeout had been scheduled,
+         * <tt>false</tt> otherwise.
+         */
+        boolean isScheduled()
+        {
+            synchronized (syncRoot)
+            {
+                return timeoutThread != null;
```

Comment:
Why do we have a synchronized block just for a null check?

Reply:
So when the transcriber is kicked we can get two MEMBER_LEFT events on memberPresenceChanged, very quickly one after another, as I'm not sure are those running in the same thread or not ... but I was seeing IllegalStateException several times (the exception thrown after isScheduled() in the code below).
The first can start creating the timeoutThread and we don't want the second to try this at the same time and get IllegalStateException... just to wait for the other to finish work before returning the result in memberPresenceChanged
```
if (!leaveTimeout.isScheduled())
                    leaveTimeout.scheduleTimeout(leaveTime);
```. Just protecting reading of that field while writing in it...
Does this make sense? Or you think we should change it somehow and keep the check near the writing of timeoutThread.

Reply:
I don't understand how this synchronized block around the null check fixes the problem that you describe. If you have two threads execute the `if`, the race is still possible. It seems to me that you need to synchronize the whole `if`.

Reply:
Ups, now I see what you mean :)

Reply:
Does it look better now?